activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -292,6 +292,8 @@ def scope
       end
 
       def null_scope?
```

Comment:
what about adding them in one line , i believe this will give the same result and easier to understand 
`reflection.has_inverse? && reflection.inverse_of.options[:optional] && owner.new_record? && !foreign_key_present?`